tests/unit/model/session_test.py

```diff
@@ -2629,17 +2632,444 @@ def test_state_normalization(
+                    )
+                ],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
+        (
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Int", "Integer", type_=rty.Integer("Integer", rty.Bounds(0, 255))
+                    ),
+                ],
+                transitions=[Transition(target=ID("End"))],
+            ),
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Int", "Integer", type_=rty.Integer("Integer", rty.Bounds(0, 255))
+                    ),
+                ],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
+        (
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("Msg", expr.Variable("E"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("Msg", expr.Variable("E"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
+        (
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("List", expr.Variable("Msg"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("List", expr.Variable("Msg"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
```

Comment:
Both test cases look very similar, but the first one seems invalid to me: `Msg'Append (E)` - appending an element to a message.

Reply:
I have removed the first one.